### ISSUE TITLE
4.0.7 release breaks {{#if @partial-block}} usage

### ISSUE THREAD

**Issue by jithin1987** [wycats/handlebars.js#1341]
```
TypeError: Cannot read property 'data' of undefined
    at Object.partialBlockWrapper (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:264:49)
    at Object.<anonymous> (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:10:33)
    at Object.eval (eval at createFunctionContext (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:10:32)
    at main (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:175:32)
    at ret (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:178:12)
    at ret (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:525:21)
    at Object.invokePartial (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:281:12)
    at Object.invokePartialWrapper [as invokePartial] (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:68:39)
    at Object.eval (eval at createFunctionContext (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:12:28)
    at main (/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:175:32)

```
Works fine with 4.0.6 release, was there any breaking changes with 4.0.7 ?

**Comment by nknapp** [comment]
No, it is not supposed to break anything. Could you provide an example that causes this error?

Am 2. Mai 2017 04:27:45 MESZ schrieb Jithin Emmanuel <notifications@github.com>:
>```
>TypeError: Cannot read property 'data' of undefined
>at Object.partialBlockWrapper
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:264:49)
>at Object.<anonymous>
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:10:33)
>at Object.eval (eval at createFunctionContext
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23),
><anonymous>:10:32)
>at main
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:175:32)
>at ret
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:178:12)
>at ret
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:525:21)
>at Object.invokePartial
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:281:12)
>at Object.invokePartialWrapper [as invokePartial]
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:68:39)
>at Object.eval (eval at createFunctionContext
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23),
><anonymous>:12:28)
>at main
>(/Users/me/app/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:175:32)
>
>```
>Works fine with 4.0.6 release, was there any breaking changes with
>4.0.7 ?
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/wycats/handlebars.js/issues/1341

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by jithin1987** [comment]
@nknapp I narrowed it down to this usage which was working in 4.0.6 but breaking with 4.0.7.

partial.handlebars
```
...
    {{#> submitcontainer}}
    <button formaction="{{deleteUrl}}" data-type="{{type}}">Delete</button>
    {{/submitcontainer}}

...
```

submitcontainer.handlebars

```
...
  {{#if @partial-block}}
  {{> @partial-block }}
  {{/if}}
...
```


4.0.7 is not handling `{{#if @partial-block}}` check. This is a working usage with 4.0.6

**Comment by nknapp** [comment]
The bug is now fixed and a new release 4.0.8 is published. I decided to publish immediately (instead of waiting for feedback) since this was only a small fix.

Thanks for your example. It helped a lot (and is the basis for the regression-test)
